coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -216,34 +220,67 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
       }
     }
 
+    sealed abstract class Consolidatable
+    object Consolidatable {
+      def all = Set[Consolidatable](Rows, Rollups)
+    }
+    case object Rows extends Consolidatable
+    case object Rollups extends Consolidatable
+
     // A series of events is consolidatable if it is an update which only
-    // changes row data, which is to say if it has the form:
+    // changes row data or rollups, which is to say if it has the form:
     //    RowsChangedPreview
     //    zero or one Truncated
     //    zero or more RowDataUpdated
     //    LastModifiedChanged
     // or
+    //    zero or more intermingled RollupCreatedOrUpdated and RollupDropped
     //    LastModifiedChanged
-    def consolidatable(it: BufferedIterator[Delogger.LogEventCompanion]): Boolean = {
+    // or
+    //    LastModifiedChanged
+    def consolidatable(it: BufferedIterator[Delogger.LogEventCompanion]): Set[Consolidatable] = {
       if(it.hasNext && it.head == Delogger.LastModifiedChanged) {
         it.next()
-        return !it.hasNext;
+        if(it.hasNext) {
+          return Set.empty
+        } else {
+          // Empty changeset, can be consolidated with anything
+          return Set(Rows, Rollups)
+        }
       }
 
-      if(!it.hasNext || it.next() != Delogger.RowsChangedPreview) {
-        return false
-      }
-      if(it.hasNext && it.head == Delogger.Truncated) {
-        it.next()
-      }
-      while(it.hasNext && it.head == Delogger.RowDataUpdated) {
-        it.next()
-      }
-      if(!it.hasNext || it.next() != Delogger.LastModifiedChanged) {
-        return false
+      if(!it.hasNext) {
+        return Set.empty
       }
 
-      !it.hasNext
+      it.next() match {
```

Comment:
👍 read through this logic and looks to me like match for the updated comment above. Only thing I'm still confused about is the "single LastModifiedChanged" log event scenario (it looks like there must be a hasNext of any type for that to work?)

Reply:
So, LastModifiedChanged should always be the last event in a version.  If it's not something's up and we bail (== don't consolidate out of caution).  What the new code at lines 242-254 is doing is saying "If there is a first event and that first event is LMC and there are no more events, we can consolidate with anything (hm 248 should be "return Consolidatable.all", I'll fix that) otherwise we can consolidate with nothing.  If there was no first event something unexpected is happening and we can consolidate with nothing".

Which means that if we get here to 256, we know that there is a first event and it is _not_ LMC so we can split down the two different types of consolidation that we're willing to perform.

Reply:
This was just the same misunderstanding on my part as above